driver/src/db_interface.rs

```diff
@@ -124,12 +124,18 @@ impl DbInterface for MongoDB {
         ];
 
         info!("Querying standing_orders: {:?}", pipeline);
-        self.client
+        let non_zero_standing_orders = self.client
             .db(models::DB_NAME)
             .collection("standing_orders")
             .aggregate(pipeline, None)?
             .map(|d| d.map(models::StandingOrder::from).map_err(DriverError::from))
-            .collect::<Result<Vec<_>, _>>()
+            .collect::<Result<Vec<_>, _>>()?;
+
+        let mut standing_orders = (0..models::NUM_RESERVED_ACCOUNTS)
+            .map(|k| models::StandingOrder::empty(k as u16))
+            .collect::<Vec<models::StandingOrder>>();
+        non_zero_standing_orders.iter().for_each(|k| standing_orders[k.account_id as usize] = k.clone());
```

Comment:
`into_iter` instead of `iter` will make it so that your `k` inside this closure is the value and not the reference. This way you don't need `k.clone()` anymore.

However, accessing the index (`k.account`) when setting an element of `Vec` is considered to happen after the value that is being set is consumed (in this case k, so k.account is accessing a value after it is consumed).
Therefore, you need to close k.account_id before using it (yet this is just cloning a single u32 instead of a full db record).

Here is the code:

```rust
non_zero_standing_orders.into_iter().for_each(|k| {
  let acc_id = k.account_id.clone() as usize;
  standing_orders[acc_id] = k;
});
```